src/test/java/microsoft/exchange/webservices/data/core/EwsUtilitiesTest.java

```diff
@@ -54,6 +56,11 @@ public void testGetBuildVersion() {
   }
 
   @Test
+  public void testParseInt() throws IllegalAccessException, ParseException, InstantiationException {
+    Assert.assertEquals(new Integer(14), EwsUtilities.parse(Integer.class, "14"));
+  }
```

Comment:
I'm happy to add more tests for the other types.

As for BigInteger etc, I can't see the tests you've included. And my changes wouldn't break that as it previously returned an Integer. All I've changed is the order of the classes in the isAssignableFrom() check